some fixes in older articles

lambda-abstraction opened the pull request.

lambda-abstraction committed:
some fixes in older articles

lambda-abstraction committed:
this was a bad idea

decorator-factory commented on docs/antipatterns/optional-attributes/index.md:
```diff
@@ -48,7 +48,7 @@ Suppose then that you come up with this solution:
 from collections.abc import Mapping
 from dataclasses import dataclass
-
+from typing import Optional
```
Maybe it would be better to replace `Optional` with `| None` in this article? It's kinda soft-deprecated

lambda-abstraction committed:
Optional[T] -> T | None, remove the older 'Optional is not optional' article

lambda-abstraction committed:
Bring back the "Optional-is-not-optional" article
Well, turns out doing changes and removing sort-of unrelated files in one commit is a bad idea.
